<?xml version="1.0"?>
<robot name="braccio_with_vision" xmlns:xacro="http://www.ros.org/wiki/xacro">

    <!-- Incluir el robot Braccio original -->
    <xacro:include filename="$(find braccio_description)/urdf/braccio.urdf.xacro" />
    
    <!-- Incluir la c치mara cenital -->
    <xacro:include filename="$(find braccio_vision)/urdf/overhead_camera.urdf.xacro" />

    <!-- Argumentos -->
    <xacro:arg name="robot_name" default="braccio" />
    <xacro:arg name="base_frame" default="world" />
    <xacro:arg name="sim" default="true" />
    <xacro:arg name="hw_test" default="false" />
    <xacro:arg name="camera_height" default="1.0" />

    <!-- A침adir la c치mara cenital al mundo -->
    <xacro:overhead_camera 
        parent_frame="$(arg base_frame)" 
        height="$(arg camera_height)" 
        camera_name="overhead_camera" />

    <!-- 
    Nota: Los objetos de prueba se generan din치micamente con object_spawner.py
    Esto permite mayor flexibilidad en posicionamiento y tipos de objetos
    -->

</robot>
